File: Beakjoon/src/Gold/Lv5/boj_13975.java
package Gold.Lv5;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.PriorityQueue;
import java.util.stream.Stream;

public class boj_13975 { // 파일 합치기 3
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int T = Integer.parseInt(br.readLine());

        for (int t = 1; t <= T; t++) {
            PriorityQueue<Long> pq = new PriorityQueue<>();
            long result = 0;
            int N = Integer.parseInt(br.readLine());
            long[] file = Stream.of(br.readLine().split(" ")).mapToLong(Long::parseLong).toArray();

            for (int i = 0; i < N; i++) {
                pq.add(file[i]);
            }

            while (pq.size() != 1) {
                long temp = pq.poll() + pq.poll();
                pq.add(temp);
                result += temp;
            }

              bw.write(result + "\n");
        }
        bw.flush();
        bw.close();
    }
}

File: Beakjoon/src/Gold/Lv5/boj_13975.kt
package Gold.Lv5

import java.util.*

fun main(): Unit = with(System.`in`.bufferedReader()) {
    val T = readLine().toInt()

    val pq = PriorityQueue<Long>()
    val answer = StringBuilder()

    for(i in 1..T){
        pq.clear()

        val K = readLine().toInt()

        val files = readLine().split(' ').map{it.toLong()}
        pq.addAll(files)

        var cost = 0L
        while(pq.size > 1){
            val min1 = pq.poll()
            val min2 = pq.poll()

            pq.offer(min1 + min2)
            cost += min1 + min2
        }

        answer.append("$cost\n")
    }
    print(answer.toString().trim())
}